pkg/scheduler/api/resource_info.go

```diff
@@ -260,6 +260,15 @@ func (r *Resource) Sub(rr *Resource) *Resource {
 	return r.sub(rr)
 }
 
+// WithoutAssertSub subtracts two Resource objects without assertion.
+func (r *Resource) WithoutAssertSub(rr *Resource) *Resource {
```

Comment:
It would be better to explain why such a function is added. For example, some resource subtraction allows negative results, while others do not.

Reply:
done